tests/unit/s2n_early_data_test.c

```diff
@@ -1250,6 +1259,7 @@ int main(int argc, char **argv)
         valid_connection->early_data_state = S2N_EARLY_DATA_ACCEPTED;
         valid_connection->early_data_bytes = 0;
         valid_connection->actual_protocol_version = S2N_TLS13;
+        EXPECT_OK(s2n_conn_choose_state_machine(valid_connection, S2N_TLS13));
```

Comment:
All the tests that I changed in this PR were ones that were failing. Essentially they were ones that needed to set the state machine to TLS13, but weren't after my change since actual_protocol_version no longer controls the state machine.
One thing I could do is remove all actual_protocol_version setting during the unit tests. But I would be worried about how big of a change that would be.

Reply:
My real concern is that there are tests that are setting actual_protocol_version, were using the TLS1.3 state machine before, but now aren't _and they're also not failing_, they're just no longer testing the right thing. A good test should verify that it's testing the right thing, but we don't always do that :\

Reply:
Sure, but how should I address that concern? Again, I can manually remove all actual_protocol_version setting in the tests, that might work. But that is an 84-file change 😢.

Reply:
Removing it wouldn't work, since tests need that if they don't do a full handshake, but maybe you could find every instance where we set actual_protocol_version and set the state machine at the same time? You'd definitely want to do that programatically, and maybe in a different PR, bc I suspect that will be a LOT of changes. And if you do that, don't forget [testlib](https://github.com/aws/s2n-tls/blob/main/tests/testlib/s2n_testlib.h#L76).

Reply:
#3756